runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactory.java

```diff
@@ -213,6 +226,12 @@ private static int getMaxEnvironmentClients(JobInfo jobInfo) {
     return maxEnvironments;
   }
 
+  private static boolean shouldLoadBalanceBundles(JobInfo jobInfo) {
```

Comment:
Would only pass in `PipelineOptions` here.

Reply:
We also need to throw here if state caching is enabled. Otherwise, we will get incorrect results because the Runner always sends the same cache token to the Harness.

Reply:
will add that upstream